### PR TITLE
feat(undelegate): use Timestamp instead of Date

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
refs: #9042

## Description

This PR primarily improves unit testing around the `.undelegate()` flow and was motivated by a failure (inability to decode `completionTime`) discovered in E2E testing. 

- Add tests for cosmos-orchestration-account staking flows using mocked IBC messages
- Use `Timestamp` (from [google/protobuf/timestamp.proto](https://buf.build/protocolbuffers/wellknowntypes/docs/657250e6a39648cbb169d079a60bd9ba:google.protobuf#google.protobuf.Timestamp)) instead of `Date` for `@agoric/cosmic-proto` codegen for better endo compatibility
- Update `undelegate()` to use Timestamp for `wakeAt()` calculation

### Testing Considerations
This PR includes additional tests for cosmos-orchestration-account flows.

### Upgrade Considerations
This is a breaking change for `@agoric/cosmic-proto`, but not for the `agoric` namespace.

### CODE REVIEW

File: packages/orchestration/src/exos/local-orchestration-account.js
Comment (0xpatrickdev) [inline packages/orchestration/src/exos/local-orchestration-account.js:92]: Using `splitRecord` here to make this forward compatible (cosmos v0.50 adds `amount: Coin` to the response: https://buf.build/cosmos/cosmos-sdk/docs/main:cosmos.staking.v1beta1#cosmos.staking.v1beta1.MsgUndelegateResponse)

Are there other places we might need to consider this approach? Should we be guarding watcher's more loosely?

File: packages/orchestration/src/cosmos-api.ts
Comment (dckc) [inline packages/orchestration/src/cosmos-api.ts]:
```suggestion
  /** deposit payment (from zoe, for example) to the account */
```

Zoe isn't the only source of payments.

File: packages/cosmic-proto/test/helpers.test-d.ts
Comment (0xpatrickdev) [inline packages/cosmic-proto/test/helpers.test-d.ts:73]: FYI. didn't seem worth checking in a patch, but happy to make one

File: packages/orchestration/test/examples/stake-ica.contract.test.ts
Comment (turadg) [inline packages/orchestration/test/examples/stake-ica.contract.test.ts]: I was reviewing by commit so I drafted a comment asking when, then I saw the fix in a subsequent PR.

I support this style of queuing up work for other commits in the PR. Consider being more specific for the reader, e.g.  `FIXME in this PR`

File: packages/orchestration/test/examples/stake-ica.contract.test.ts
Comment (turadg) [inline packages/orchestration/test/examples/stake-ica.contract.test.ts]: typo

File: packages/orchestration/test/examples/stake-ica.contract.test.ts
Comment (turadg) [inline packages/orchestration/test/examples/stake-ica.contract.test.ts:185]: our composite "address" object reads so weirdly. I hope we can come up with a better name.

File: packages/orchestration/src/exos/local-orchestration-account.js
Comment (0xpatrickdev) [reply packages/orchestration/src/exos/local-orchestration-account.js:92]: I guess the local-orch-account is less likely to get an unexpected upgrade to v0.50, more the cosmos-orch-account. We do not have something similar in those interface guards since the value is still a base64 byte string at that point.

Comment (turadg) [reply packages/orchestration/src/exos/local-orchestration-account.js:92]: Good question tho. Thanks for raising and resolving.

File: packages/orchestration/src/typeGuards.js
Comment (turadg) [inline packages/orchestration/src/typeGuards.js:108]: excellent comment

File: packages/orchestration/src/utils/time.js
Comment (turadg) [inline packages/orchestration/src/utils/time.js:63]: reduction in code affirms your idea to switch to Timestamp

File: packages/orchestration/src/examples/sendAnywhere.contract.js
Comment (dckc) [inline packages/orchestration/src/examples/sendAnywhere.contract.js:72]: :clap:

File: packages/orchestration/test/ibc-mocks.ts
Comment (turadg) [inline packages/orchestration/test/ibc-mocks.ts]: it's only five because of `UNBOND_PERIOD_SECONDS`. if someone changes that they won't know to change this comment.

the 0+ is thoughtful, but I don't think the "since unix epoch" is relevant to the test

please simplify and make more explicit what it gets
```suggestion
const getUnbondingTime = () => new Date(Number(UNBOND_PERIOD_SECONDS * MILLISECONDS_PER_SECOND));
```

**Review by turadg — APPROVED** [review]
Thanks for the clearly described separate commits.

Only minor issues. Approving assuming they'll be addressed before merge.

File: packages/orchestration/src/orchestration-api.ts
Comment (dckc) [inline packages/orchestration/src/orchestration-api.ts:179]: cc @dtribble  - change to `orchestration-api.ts`

File: packages/orchestration/test/examples/stake-ica.contract.test.ts
Comment (dckc) [reply packages/orchestration/test/examples/stake-ica.contract.test.ts:185]: cc @mitdralla

File: packages/orchestration/test/staking-ops.test.ts
Comment (dckc) [inline packages/orchestration/test/staking-ops.test.ts:102]: here's hoping we manage to completely replace the mock stuff here with using ibc-mocks

File: packages/orchestration/src/exos/local-orchestration-account.js
Comment (dckc) [inline packages/orchestration/src/exos/local-orchestration-account.js:193]: :clap: good to see the TODOs getting to-done.

File: packages/orchestration/test/examples/stake-ica.contract.test.ts
Comment (dckc) [inline packages/orchestration/test/examples/stake-ica.contract.test.ts]: :clap:

File: packages/orchestration/test/exos/local-orchestration-account-kit.test.ts
Comment (dckc) [inline packages/orchestration/test/exos/local-orchestration-account-kit.test.ts:149]: you can test that it actually waited that long by racing with `timer.wakeAt(expectedFullfillment - 1n)`

**Review by dckc — APPROVED** [review]
very nicely done!

File: packages/orchestration/src/orchestration-api.ts
Comment (dckc) [reply packages/orchestration/src/orchestration-api.ts:179]: cc @mitdralla too

**Review by dckc — COMMENTED** [review]
oops; some comments pending from another window

File: packages/orchestration/test/exos/local-orchestration-account-kit.test.ts
Comment (0xpatrickdev) [reply packages/orchestration/test/exos/local-orchestration-account-kit.test.ts:149]: Updated to include, ty. Also realized I don't need to provide a brand to the zone manual timer.

File: packages/orchestration/test/staking-ops.test.ts
Comment (0xpatrickdev) [reply packages/orchestration/test/staking-ops.test.ts:102]: Agree. It seems there's parity here with `stake-ica.test.ts` minus the wallet offers.

Will +1 that we should be able to use a macro and test LocalOrchestrationAccount in the same flow. Currently there are two small blockers to that:
  - LOA `delegate` and `undelegate` should accept a ChainAddress obj instead of a string
  - LOA `undelegate` should accept an array of delegations

File: packages/orchestration/test/ibc-mocks.ts
Comment (0xpatrickdev) [reply packages/orchestration/test/ibc-mocks.ts]: Simplified but kept the return type a Timestamp

File: packages/orchestration/test/examples/stake-ica.contract.test.ts
Comment (0xpatrickdev) [reply packages/orchestration/test/examples/stake-ica.contract.test.ts]: Acknowledged for the future

File: packages/orchestration/src/orchestration-api.ts
Comment (0xpatrickdev) [reply packages/orchestration/src/orchestration-api.ts:179]: Similar story to: https://github.com/Agoric/agoric-sdk/pull/9591#discussion_r1659007324

Comment (dtribble) [reply packages/orchestration/src/orchestration-api.ts:179]: lgtm

File: packages/orchestration/test/ibc-mocks.ts
Comment (turadg) [reply packages/orchestration/test/ibc-mocks.ts]: oh good. I made the suggestion on an earlier commit and when I pulled it to the PR level I forgot to adjust the return type

**Pull request merged**
